kyaml/openapi/openapi.go

```diff
@@ -368,27 +391,68 @@ func initSchema() {
 		}
 
 		// parse the swagger, this should never fail
-		if _, err := parse(kubernetesapi.MustAsset(kubernetesAPIAssetName)); err != nil {
+		if err := parse(kubernetesapi.MustAsset(kubernetesAPIAssetName)); err != nil {
 			// this should never happen
 			panic(err)
 		}
 
-		if _, err := parse(kustomizationapi.MustAsset(kustomizationAPIAssetName)); err != nil {
+		if err := parse(kustomizationapi.MustAsset(kustomizationAPIAssetName)); err != nil {
 			// this should never happen
 			panic(err)
 		}
 	})
 }
 
 // parse parses and indexes a single json schema
-func parse(b []byte) (*spec.Schema, error) {
-	var sc spec.Schema
+func parse(b []byte) error {
+	var swagger spec.Swagger
 
-	if err := sc.UnmarshalJSON(b); err != nil {
-		return nil, errors.Wrap(err)
+	if err := swagger.UnmarshalJSON(b); err != nil {
+		return errors.Wrap(err)
+	}
+	AddDefinitions(swagger.Definitions)
+	findNamespaceability(swagger.Paths)
+
+	return nil
+}
+
+// findNamespaceability looks at the api paths for the resource to determine
+// if it is cluster-scoped or namespace-scoped. The gvk of the resource
+// for each path is found by looking at the x-kubernetes-group-version-kind
+// extension. If a path exists for the resource that contains a namespace path
+// parameter, the resource is namespace-scoped.
+func findNamespaceability(paths *spec.Paths) {
+	if globalSchema.namespaceabilityByResourceType == nil {
+		globalSchema.namespaceabilityByResourceType = make(map[yaml.TypeMeta]bool)
+	}
+
+	if paths == nil {
+		return
+	}
+
+	for path, pathInfo := range paths.Paths {
+		if pathInfo.Get == nil {
+			continue
+		}
+		gvk, found := pathInfo.Get.VendorExtensible.Extensions[kubernetesGVKExtensionKey]
+		if !found {
+			continue
+		}
+		typeMeta, found := toTypeMeta(gvk)
+		if !found {
+			continue
+		}
+
+		if strings.Contains(path, "namespaces/{namespace}") {
```

Comment:
Yeah, you have two options:
1. Look for that specific pattern in the path, or
2. A path parameter like this: 
```
"in": "path",
"name": "namespace",
"required": true,
"type": "string",
```
3. Or both.

Reply:
Yeah, I think both approaches should work equally well. The only tricky part is that even namespaced resources will have some paths that doesn't include the namespace path parameter (like `/apis/apps/v1/deployments`). But the code does account for this.

Reply:
Right, that's fine! If you feel confident that this is good enough, go for it!
/lgtm